internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -100,6 +110,135 @@ func validateSidecarMountPoints(in []manifest.SidecarMountPoint) error {
 	return nil
 }
 
+func validateSidecarDependsOn(in manifest.SidecarConfig, sidecarName string, sidecars map[string]*manifest.SidecarConfig, manifestName string) error {
+	if in.DependsOn == nil {
+		return nil
+	}
+
+	for name, status := range in.DependsOn {
+		// don't let a sidecar depend on itself
+		if name == sidecarName {
+			return errCircularDependency
+		}
+		// status must be one of < start | complete | success >
+		if !isValidStatus(status) {
+			return errInvalidDependsOnStatus
+		}
+		// essential containers must have 'start' as a status
+		if name == manifestName || sidecars[name].Essential == nil || *sidecars[name].Essential {
+			if status != "start" {
+				return errEssentialContainerStatus
+			}
+		}
+	}
+
+	return nil
+}
+
+func validateNoCircularDependencies(sidecars map[string]*manifest.SidecarConfig, img manifest.Image, manifestName string) error {
+	// don't let anything circularly depend on each other
+	dependencies := buildDependencyGraph(sidecars, img, manifestName)
+
+	// don't let invalid containers happen
+	if dependencies == nil {
```

Comment:
We should probably throw the error earlier rather than implicitly refer the `nil` return value to be this value type.

Reply:
Adjusted